src/librustc/ty/wf.rs

```diff
@@ -47,8 +50,9 @@ pub fn trait_obligations<'a, 'tcx>(
     body_id: hir::HirId,
     trait_ref: &ty::TraitRef<'tcx>,
     span: Span,
+    item: Option<&'tcx hir::Item>,
```

Comment:
I don't think passing around references to the HIR fits `ty::wf`, this can just be an `Option<DefId>` and should probably be named something more suggestive.